test/test_sparse_csr.py

```diff
@@ -3468,6 +3469,125 @@ def test_triton_bsr_dense_bmm_error_messages(self, device, dtype):
             out = torch.rand(32, 32, 2, dtype=dtype, device=device).transpose(0, -1)
             bsr_dense_mm(lhs, rhs, out=out)
 
+    @parametrize("block_size", [16, 32, 64])
+    @onlyCUDA
+    @skipIfRocm
+    @dtypes(torch.half, torch.bfloat16, torch.float)
+    @dtypesIfCUDA(torch.half, *[torch.bfloat16] if SM80OrLater else [], torch.float)
+    @unittest.skipIf(IS_FBCODE and IS_REMOTE_GPU, "Test requires Triton")
+    def test_triton_sampled_addmm(self, device, dtype, block_size):
+        from functools import partial
+        from torch.sparse._triton_ops import sampled_addmm, broadcast_batch_dims_bsr, tile_to_blocksize
+
+        # Note that each value in a non-zero block is in range block_size * [low^2, high^2).
+        tensor = partial(make_tensor, device=device, dtype=dtype, low=0.3, high=1.2)
+
+        # NOTE: batch dims with zero sizes are not supported in `to_sparse_bsr`.
+        batches = [(), (2,), (2, 2)]
+        size = [128, 256, 0]
+
+        def sampled_addmm_ref(input, mat1, mat2, alpha, beta):
```

Comment:
Could another way to implement this be `torch.addmm(input.to_dense(), mat1, mat2, alpha, beta).sparse_mask(input.to_sparse()).to_sparse_bsr(input.values().shape[-2:])`?

Sure it's less efficient, but a simpler reference implementation could give us more confidence?

Reply:
We test against CSR from `sparse.sampled_addmm` for more confidence. The proposed solution also only works for non-batched inputs, so, for simplicity, we still need to loop over batches...

Reply:
Yes, testing against CSR with float would work too. That's even simpler.